cli/src/diff_util.rs

```diff
@@ -749,6 +759,109 @@ pub fn show_diff_summary(
     })
 }
 
+struct DiffStat {
+    path: RepoPath,
+    added: usize,
+    removed: usize,
+}
+
+fn get_diff_stat<'content>(
+    repo_path: &RepoPath,
+    left_content: &'content [u8],
+    right_content: &'content [u8],
+) -> DiffStat {
+    let hunks = unified_diff_hunks(left_content, right_content, 0);
+    let mut adds = 0;
+    let mut dels = 0;
+    for hunk in hunks {
+        for (line_type, _content) in hunk.lines {
+            match line_type {
+                DiffLineType::Context => {}
+                DiffLineType::Removed => dels += 1,
+                DiffLineType::Added => adds += 1,
+            }
+        }
+    }
+    DiffStat {
+        path: repo_path.clone(),
+        added: adds,
+        removed: dels,
+    }
+}
+
+pub fn show_diff_stat(
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    tree_diff: TreeDiffIterator,
+) -> Result<(), CommandError> {
+    let mut stats: Vec<DiffStat> = vec![];
+    let mut max_path_length = 0;
+    let mut max_diffs = 0;
+    for (repo_path, diff) in tree_diff {
+        match diff {
+            tree::Diff::Modified(left, right) => {
+                let left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                let right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+                let stat = get_diff_stat(&repo_path, &left_content, &right_content);
+                stats.push(stat);
+            }
+            tree::Diff::Added(right) => {
+                let right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+                let stat = get_diff_stat(&repo_path, &[], &right_content);
+                stats.push(stat);
+            }
+            tree::Diff::Removed(left) => {
+                let left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                let stat = get_diff_stat(&repo_path, &left_content, &[]);
+                stats.push(stat);
+            }
+        }
+        max_path_length = max(max_path_length, repo_path.to_internal_file_string().len());
+        max_diffs = stats
+            .iter()
+            .map(|stat| max(max_diffs, stat.added + stat.removed))
+            .max()
+            .unwrap_or(0);
```

Comment:
If I'm reading this correctly, it's O(n^2). I suppose you meant to put it outside the loop and write it like this:

```rust
let max_diffs = stats.iter().map(|stat| stat.added + stat.removed).max().unwrap_or(0);
```

Reply:
lol, yeah... I changed how I wrote it a few times and ended up with both calls to `max()`... good catch....

Reply:
Fixed.